model/src/main/java/com/cloudera/nav/sdk/model/entities/PigOperation.java

```diff
@@ -0,0 +1,35 @@
+package com.cloudera.nav.sdk.model.entities;
+
+import com.cloudera.nav.sdk.model.SourceType;
+import com.cloudera.nav.sdk.model.annotations.MClass;
+import com.google.common.base.Strings;
+
+@MClass(model="pig_operation", validTypes = {EntityType.OPERATION})
+public class PigOperation extends Entity {
+
+  public PigOperation() {
+    setSourceType(SourceType.PIG);
+    setEntityType(EntityType.OPERATION);
+  }
+
+  public PigOperation(String logicalPlanHash, String jobName) {
+    this();
+    setLogicalPlanHash(logicalPlanHash);
+    setJobName(jobName);
+  }
+
+  public PigOperation(String id) {
+    this();
+    setIdentity(id);
+  }
+
+  @Override
+  public void validateEntity() {
+    if (Strings.isNullOrEmpty(this.getIdentity()) &&
+        (Strings.isNullOrEmpty(this.getJobName()) ||
+        Strings.isNullOrEmpty(this.getLogicalPlanHash())))
+      throw new IllegalArgumentException(
```

Comment:
Use Preconditions here?

Reply:
Preconditions checks for an argument to be true. Something like Preconditions.checkArgument() and it throws the error if the expression is false. We can put Preconditions here but it will just add more of !, which will make these conditions look messy.